core-bundle/src/Resources/contao/modules/ModulePersonalData.php

```diff
@@ -131,34 +131,34 @@ protected function compile()
 		// Build the form
 		foreach ($this->editable as $field)
 		{
-			$arrData = &$GLOBALS['TL_DCA']['tl_member']['fields'][$field];
+			$arrData = $GLOBALS['TL_DCA']['tl_member']['fields'][$field] ?? array();
```

Comment:
I think this line needs to be reverted because the lines below change `$arrData` and this does now no longer affect the `$GLOBALS['TL_DCA']` array.

Reference to a non-existing array key should not trigger a notice or warning.

Reply:
There are multiple places where we do `$arrData = $GLOBALS['TL_DCA']…`. This one was the only one where we assigned a reference instead of copying the array, so I figured we should adjust it to be consistent. WDYT?

Reply:
We could also always assign a reference (`$arrData = &$GLOBALS['TL_DCA']…`), so changes to `$arrData` are also available in `$GLOBALS['TL_DCA']`. However, I do not think that this is what we want:

https://github.com/contao/contao/blob/6f3a683cd4679fb0a73eefbdce9304de98c344c3/core-bundle/src/Resources/contao/modules/ModulePersonalData.php#L134-L140

The input type change here is only necessary in this particular module. It actually should not be applied to the original array.

Reply:
You are right, using a reference here seems to be wrong.